lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -172,17 +172,52 @@ TruncatedDistribution * TruncatedDistribution::clone() const
   return new TruncatedDistribution(*this);
 }
 
+Distribution TruncatedDistribution::dispatchTruncation(const Collection<Distribution> & distributions) const
+{
+  const UnsignedInteger size = distributions.getSize();
+  Collection<Distribution> newBlocks(size);
+  UnsignedInteger startIndex = 0;
+  for (UnsignedInteger i = 0; i < size; ++ i)
+  {
+    Indices blockIndices(distributions[i].getDimension());
+    blockIndices.fill(startIndex);
+    newBlocks[i] = TruncatedDistribution(distributions[i], bounds_.getMarginal(blockIndices));
```

Comment:
Here we could store the simplified version of the marginal as we do in PointConditionalDistribution